### PR TITLE
Provides true path validation.

### PR DESCRIPTION
**Pull request by brad-jones**
Has changes described here: https://github.com/carlansley/swagger2/issues/18

Tests have been updated to pass but no new path specific tests added.

### CODE REVIEW

File: src/validate.ts [hunk 1]
```diff
@@ -88,6 +88,7 @@ function validate(value: any, schema: CompiledDefinition): ValidationError | und
 }
 
 export function request(compiledPath: CompiledPath | undefined,
+                        pathParameters: { [name: string]: any },
```
Comment (carlansley) [inline src/validate.ts]: This should go at the end of the parameter list, and be optional, to make sure we don't break existing code.

File: dist/validate.d.ts [hunk 2]
```diff
@@ -6,5 +6,7 @@ export interface ValidationError {
     expected: any;
     error?: any;
 }
-export declare function request(compiledPath: CompiledPath | undefined, method: string, query?: any, body?: any, headers?: any): ValidationError[] | undefined;
+export declare function request(compiledPath: CompiledPath | undefined, pathParameters: {
```
Comment (carlansley) [inline dist/validate.d.ts]: Can you revert all the dist files from the PR.  I'll recreate when I publish a new version.

**Comment by brad-jones** [comment]
Suggested changes have been made, also added in 2 new test cases for path validation.

**Comment by carlansley** [comment]
Thanks for the contribution!

**Pull request merged**